### PR TITLE
feat: add starknet RPC estimate_fee

### LINKED ISSUE #156: feat: Initial Starknet RPC "estimate_fee" call
# Description

Implement "estimate_fee" Starknet RPC call

Starknet-RS Implementation
https://github.com/xJonathanLEI/starknet-rs/blob/a6d864fb0b02a9b530283148047ab5027ce4d295/starknet-providers/src/jsonrpc/mod.rs#L310

# Acceptance Criteria

- Beerus can query a Starknet's block number
- Associated cli command
- Associated rest api wrapper
- Units tests
- Integration tests

### PR DESCRIPTION
**Pull request by ftupas**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: Fix #156 

# What is the new behavior?

- add estimate_fee to core
- add estimate_fee to CLI
- add estimate_fee to API

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

# Other information

### CODE REVIEW

File: beerus_cli/src/runner.rs [hunk 1]
```diff
@@ -186,6 +186,19 @@ pub async fn run(beerus: BeerusLightClient, cli: Cli) -> Result<CommandResponse>
...
+                broadcasted_transaction,
+            } => {
+                starknet::query_starknet_estimate_fee(
+                    beerus,
+                    block_id.to_string(),
+                    block_id_type.to_string(),
+                    broadcasted_transaction.to_string(),
+                )
+                .await
+            }
```
Comment (ftupas) [inline beerus_cli/src/runner.rs:201]: This is what this call looks like

```bash
source .env && ./target/release/beerus_cli starknet query-estimate-fee \
--block-id-type tag \
--block-id latest \
--broadcasted-transaction "{\"type\":\"INVOKE\",\"max_fee\":\"0x0\",\"version\":\"0x1\",\"signature\":[\"0x156a781f12e8743bd07e20a4484154fd0baccee95d9ea791c121c916ad44ee0\",\"0x7228267473c670cbb86a644f8696973db978c51acde19431d3f1f8f100794c6\"],\"nonce\":\"0x0\",\"sender_address\":\"0x5b5e9f6f6fb7d2647d81a8b2c2b99cbc9cc9d03d705576d7061812324dca5c0\",\"calldata\":[\"0x1\",\"0x7394cbe418daa16e42b87ba67372d4ab4a5df0b05c6e554d158458ce245bc10\",\"0x2f0b3c5710379609eb5495f1ecd348cb28167711b73609fe565a72734550354\",\"0x0\",\"0x3\",\"0x3\",\"0x5b5e9f6f6fb7d2647d81a8b2c2b99cbc9cc9d03d705576d7061812324dca5c0\",\"0x3635c9adc5dea00000\",\"0x0\"]}"
```

This is the output

```
FeeEstimate { gas_consumed: 5194, gas_price: 33106896397, overall_fee: 171957219886018 }
```

File: beerus_rest_api/src/api/starknet/endpoints.rs [hunk 2]
```diff
@@ -279,6 +279,33 @@ pub async fn query_starknet_syncing(
...
+pub async fn get_estimate_fee(
+    beerus: &State<BeerusLightClient>,
+    block_id_type: String,
+    block_id: String,
+    broadcasted_transaction: String,
+) -> ApiResponse<QueryEstimateFeeResponse> {
+    ApiResponse::from_result(
+        get_estimate_fee_inner(beerus, block_id_type, block_id, broadcasted_transaction).await,
+    )
+}
```
Comment (ftupas) [inline beerus_rest_api/src/api/starknet/endpoints.rs:307]: This is what the call looks like, `broadcasted_transaction` is JSON URL-encoded

```bash
curl --silent \
http://127.0.0.1:8000/starknet/fee\?broadcasted_transaction\=%7B%22type%22%3A%22INVOKE%22%2C%22max_fee%22%3A%220x0%22%2C%22version%22%3A%220x1%22%2C%22signature%22%3A%5B%220x156a781f12e8743bd07e20a4484154fd0baccee95d9ea791c121c916ad44ee0%22%2C%220x7228267473c670cbb86a644f8696973db978c51acde19431d3f1f8f100794c6%22%5D%2C%22nonce%22%3A%220x0%22%2C%22sender_address%22%3A%220x5b5e9f6f6fb7d2647d81a8b2c2b99cbc9cc9d03d705576d7061812324dca5c0%22%2C%22calldata%22%3A%5B%220x1%22%2C%220x7394cbe418daa16e42b87ba67372d4ab4a5df0b05c6e554d158458ce245bc10%22%2C%220x2f0b3c5710379609eb5495f1ecd348cb28167711b73609fe565a72734550354%22%2C%220x0%22%2C%220x3%22%2C%220x3%22%2C%220x5b5e9f6f6fb7d2647d81a8b2c2b99cbc9cc9d03d705576d7061812324dca5c0%22%2C%220x3635c9adc5dea00000%22%2C%220x0%22%5D%7D\&block_id\=latest\&block_id_type\=tag | jq
```

This is the output
```json
{
  "gas_consumed": "5194",
  "gas_price": "22630904152",
  "overall_fee": "117544916165488"
}
```

Comment (drspacemn) [reply beerus_rest_api/src/api/starknet/endpoints.rs:327]: tested looks great

File: beerus_cli/src/runner.rs (hunk 1 above)
Comment (drspacemn) [reply beerus_cli/src/runner.rs:204]: tested looks great

**Review by drspacemn — APPROVED** [review]
ran provided integration tests, looks great!

**Pull request merged**